### PR TITLE
Add initial database schema

### PR DESCRIPTION
**Pull request by johnwatson484**

### CODE REVIEW

File: app/data/models/calculation.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module.exports = (sequelize, DataTypes) => {
+  const claim = sequelize.define('calculation', {
```
Comment (marctemp) [inline app/data/models/calculation.js]: change to `calculation`

File: app/data/models/calculation.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  claim.associate = function (models) {
+    claim.hasMany(models.funding, {
+      foreignKey: 'calculationId',
+      as: 'funding'
+    })
+    claim.belongsTo(models.organisation, {
+      foreignKey: 'sbi',
+      as: 'organisations'
+    })
+    return claim
```
Comment (marctemp) [inline app/data/models/calculation.js]: this needs moved outside this function -- same for all models

File: app/data/models/funding.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  {
+    tableName: 'fundings',
+    freezeTableName: true,
+    timestamps: false
+  })
+  funding.associate = function (models) {
+    funding.belongsTo(models.claim, {
+      foreignKey: 'calculationId',
+      as: 'claim'
```
Comment (marctemp) [inline app/data/models/funding.js]: calculations

File: app/data/models/funding.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    areaClaimed: DataTypes.DECIMAL,
+    rate: DataTypes.DECIMAL
+  },
+  {
+    tableName: 'fundings',
+    freezeTableName: true,
+    timestamps: false
+  })
+  funding.associate = function (models) {
+    funding.belongsTo(models.claim, {
```
Comment (marctemp) [inline app/data/models/funding.js]: models.calculation

File: app/data/models/organisation.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    name: DataTypes.STRING,
+    updated: DataTypes.DATE
+  },
+  {
+    tableName: 'organisations',
+    freezeTableName: true,
+    timestamps: false
+  })
+  organisation.associate = function (models) {
+    organisation.hasMany(models.claim, {
```
Comment (marctemp) [inline app/data/models/organisation.js]: calculation

File: app/data/models/organisation.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      foreignKey: 'sbi',
+      as: 'claims'
```
Comment (marctemp) [inline app/data/models/organisation.js]: calculations

File: changelog/db.changelog-1.1.xml [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      <column name="emailAddress" type="VARCHAR(127)" />
+      <column name="frn" type="BIGINT" />
+      <column name="name" type="VARCHAR(127)" />
+      <column name="updated" type="TIMESTAMP" />
+    </createTable>
+  </changeSet>
+  <changeSet author="John Watson" id="3">
+    <createTable tableName="calculations">
+      <column name="calculationId" type="INTEGER">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="calculationId_pkey" />
```
Comment (marctemp) [inline changelog/db.changelog-1.1.xml:26]: calculationsId_pkey

Comment (marctemp) [inline changelog/db.changelog-1.1.xml]: calculationsId_pkey

File: changelog/db.changelog-1.1.xml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  </changeSet>
+  <changeSet author="John Watson" id="4">
+    <createTable tableName="fundings">
+      <column name="fundingId" type="INTEGER" autoIncrement="true" >
+        <constraints nullable="false" primaryKey="true" primaryKeyName="fundingsId_pkey" />
+      </column>
+      <column name="fundingCode" type="VARCHAR(10)" />
+      <column name="calculationId" type="INTEGER" />
+      <column name="areaClaimed" type="DECIMAL(19,4)" />
+      <column name="rate" type="DECIMAL(18,6)" />
```
Comment (marctemp) [inline changelog/db.changelog-1.1.xml]: tht's even more wild tht Cross Border's area size and precision 😆

File: changelog/db.changelog-1.1.xml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    </createTable>
+  </changeSet>
+  <changeSet author="John Watson" id="5">
+    <addForeignKeyConstraint baseColumnNames="calculationId" baseTableName="fundings" constraintName="fk_calculation_calculationId" referencedColumnNames="calculationId" referencedTableName="calculations" />
```
Comment (marctemp) [inline changelog/db.changelog-1.1.xml]: fk_calculations_calculationId

File: app/data/models/calculation.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+module.exports = (sequelize, DataTypes) => {
```
Comment (marctemp) [inline app/data/models/calculation.js:1]: Suggestion: have sequelize models and changelog database creation fields in alphabetical order with foreign keys appearing at the top in alpha order

Comment (johnwatson484) [reply app/data/models/calculation.js:1]: I tend to order them by most important and grouped by similar.  eg when looking at the calculation the first columns are what the calculation is and who it's for, then the later ones are the lower level detail.

However, both options kind of become redundant in the db itself as the postgres/liquibase combo doesn't support placement of new columns when there are changes.

File: changelog/db.changelog-1.1.xml (hunk 8 above)
Comment (johnwatson484) [reply changelog/db.changelog-1.1.xml]: Indeed!  So this one is taken from the BPS Claim Statement Generator.  Some of the rates of reductions are calculated at six decimal places so have replicated the same so we can cover those schemes too.

18 might be too big though in hindsight as we know the rate isn't that high.  Will drop down

File: changelog/db.changelog-1.1.xml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    <createTable tableName="organisations">
+      <column name="sbi" type="INTEGER">
+        <constraints nullable="false" primaryKey="true" primaryKeyName="organisationsId_pkey" />
+      </column>
+      <column name="addressLine1" type="VARCHAR(127)" />
+      <column name="addressLine2" type="VARCHAR(127)" />
+      <column name="addressLine3" type="VARCHAR(127)" />
+      <column name="city" type="VARCHAR(127)" />
+      <column name="county" type="VARCHAR(127)" />
+      <column name="postCode" type="VARCHAR(8)" />
```
Comment (marctemp) [inline changelog/db.changelog-1.1.xml]: `postcode`

File: changelog/db.changelog.xml [hunk 12]
```diff
@@ -5,4 +5,5 @@
   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
                       http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.8.xsd">  
   <include  file="changelog/db.changelog-1.0.xml"/>
+  <include  file="changelog/db.changelog-1.1.xml"/>
```
Comment (marctemp) [inline changelog/db.changelog.xml:8]: dbl spaces hurt baby penguins

**Review by marctemp — APPROVED** [review]
ALLL GOODDDD

**Pull request merged**